src/main/io/ledstrip.c

```diff
@@ -623,6 +624,36 @@ static void applyLedWarningLayer(bool updateNow, timeUs_t *timer)
 }
 
 #ifdef USE_VTX_COMMON
+static uint8_t getColorByVtxFrequency(const uint16_t freq)
+{
+    if (freq < VTX_SETTINGS_MIN_FREQUENCY_MHZ || VTX_SETTINGS_MAX_FREQUENCY_MHZ < freq) {
+        return COLOR_BLACK;
+    }
+    // Freqs are divided to match Raceband channels
+    if (freq <= 5672) {
+        return COLOR_WHITE;
+    }
+    if (freq <= 5711) {
+        return COLOR_RED;
+    }
+    if (freq <= 5750) {
+        return COLOR_ORANGE;
+    }
+    if (freq <= 5789) {
+        return COLOR_YELLOW;
+    }
+    if (freq <= 5829) {
+        return COLOR_GREEN;
+    }
+    if (freq <= 5867) {
+        return COLOR_BLUE;
+    }
+    if (freq <= 5906) {
+        return COLOR_DARK_VIOLET;
+    }
+    return COLOR_DEEP_PINK;
+}
+
```

Comment:
I think it would be clearer to map frequency to channel and then use channel to lookup colour in a table. Actually, the frequencies should be pulled from a table too.

Reply:
was  this resolved?

Reply:
No, forgot this... I will modify asap

Reply:
hmm, @SteveCEvans which kind of freq to channel mapping are you thinking? VTX channel cannot be used directly here. E.g. F2 (5760MHz) should be mapped to index `COLOR_YELLOW` which is color index 4 (as it was originally and didn't want to change the legacy functionality)... 
`COLOR_` itself is already an index.

Reply:
rebased and a proposal of the lookup table added. Please check.

Reply:
@MacDaddyFPV works for me and friends of mine (I picked same for even 4.2, 4.3 and 4.4 versions and all ok). Please, make sure the settings are set correctly:
`set ledstrip_profile = RACE` 
`set ledstrip_race_color = BLACK`

Reply:
Lookup looks OK to me